src/PrestaShopBundle/Entity/Shipment.php

```diff
@@ -0,0 +1,260 @@
+     */
+    private int $carrierId;
+
+    /**
+     * @ORM\Column(name="id_delivery_address", type="integer")
+     */
+    private int $deliveryAddressId;
+
+    /**
+     * @ORM\Column(name="shipping_cost_tax_excl", type="float")
+     */
+    private float $shippingCostTaxExcluded;
+
+    /**
+     * @ORM\Column(name="shipping_cost_tax_incl", type="float")
+     */
+    private float $shippingCostTaxIncluded;
+
+    /**
+     * @ORM\Column(name="packed_at", type="datetime", nullable=true)
+     */
+    private ?DateTime $packedAt;
+
+    /**
+     * @ORM\Column(name="shipped_at", type="datetime", nullable=true)
+     */
+    private ?DateTime $shippedAt;
+
+    /**
+     * @ORM\Column(name="delivered_at", type="datetime", nullable=true)
+     */
+    private ?DateTime $deliveredAt;
+
+    /**
+     * @ORM\Column(name="tracking_number", type="string", nullable=true)
+     */
+    private ?string $trakingNumber;
+
+    /**
+     * @ORM\Column(name="created_at", type="datetime", nullable=false)
+     */
+    private DateTime $createdAt;
+
+    /**
+     * @ORM\Column(name="updated_at", type="datetime", nullable=false)
+     */
+    private DateTime $updatedAt;
+
+    /**
+     * @var Collection<ShipmentProduct>
+     *
+     * @ORM\OneToMany(targetEntity="PrestaShopBundle\Entity\ShipmentProduct", mappedBy="shipment")
+     */
+    private Collection $products;
+
+    public function __construct()
+    {
+        $this->createdAt = new DateTime('now');
+        $this->updatedAt = new DateTime('now');
+        $this->products = new ArrayCollection();
+    }
+
+    public function getProducts(): Collection
+    {
+        return $this->products;
+    }
+
+    public function getUpdatedAt()
+    {
+        return $this->updatedAt;
+    }
+
+    public function getCreatedAt()
+    {
+        return $this->createdAt;
+    }
+
+    public function getId(): int
+    {
+        return $this->id;
+    }
+
+    public function getOrderId(): int
+    {
+        return $this->orderId;
+    }
+
+    public function getCarrierId(): int
+    {
+        return $this->carrierId;
+    }
+
+    public function getDeliveryAddressId(): int
+    {
+        return $this->deliveryAddressId;
+    }
+
+    public function getShippingCostTaxExcluded(): float
+    {
+        return $this->shippingCostTaxExcluded;
+    }
+
+    public function getShippingCostTaxIncluded(): float
+    {
+        return $this->shippingCostTaxIncluded;
+    }
+
+    public function getPackedAt(): ?DateTime
+    {
+        return $this->packedAt;
+    }
+
+    public function getShippedAt(): ?DateTime
+    {
+        return $this->shippedAt;
+    }
+
+    public function getDeliveredAt(): ?DateTime
+    {
+        return $this->deliveredAt;
+    }
+
+    public function getTrakingNumber(): ?string
+    {
+        return $this->trakingNumber;
+    }
+
+    public function setId(int $id): self
+    {
+        $this->id = $id;
+
+        return $this;
+    }
```

Comment:
```suggestion
```
Doctrine entities usually don't need a setter for the ID, it is already handled by Doctrine